packages/material-ui/src/Grid/Grid.js

```diff
@@ -25,29 +29,12 @@ function getOffset(val) {
   return `${parse}${String(val).replace(String(parse), '') || 'px'}`;
 }
 
-// Duplicated with Stack.js
-function resolveBreakpointValues({ values, base }) {
-  const keys = Object.keys(base);
-
-  if (keys.length === 0) {
-    return values;
-  }
-
-  let previous;
-
-  return keys.reduce((acc, breakpoint) => {
-    if (typeof values === 'object') {
-      acc[breakpoint] = values[breakpoint] != null ? values[breakpoint] : values[previous];
-    } else {
-      acc[breakpoint] = values;
-    }
-    previous = breakpoint;
-    return acc;
-  }, {});
-}
-
 function generateGrid(globalStyles, theme, breakpoint, styleProps) {
-  const size = styleProps[breakpoint];
+  const size = getBreakpointValue({
+    data: styleProps,
+    breakpoint,
+    breakpointKeys: theme.breakpoints.keys,
+  });
 
```

Comment:
Are you sure about this tradeoff? The intent of adding

<img width="727" alt="Capture d’écran 2021-06-07 à 12 35 02" src="https://user-images.githubusercontent.com/3165635/121002617-d05c7080-c78c-11eb-8542-87cc4d7a233a.png">

Was to delay the problem to later, see if we really need to pay the cost. Here, it seems that it leads to generating a lot more CSS and spending more time to compute it. I think that it would be great to benchmark to see if this has a real impact. In the unknown, I went on assuming that it would have.

Reply:
Before
https://next.material-ui.com/components/grid/#basic-grid

<img width="1150" alt="Capture d’écran 2021-06-07 à 12 40 17" src="https://user-images.githubusercontent.com/3165635/121003201-8d4ecd00-c78d-11eb-9801-002582af1fdb.png">

After 
https://deploy-preview-26590--material-ui.netlify.app/components/grid/#basic-grid

<img width="1069" alt="Capture d’écran 2021-06-07 à 12 39 36" src="https://user-images.githubusercontent.com/3165635/121003209-8f189080-c78d-11eb-9855-e244a2088e28.png">